tck/src/main/java/ee/jakarta/tck/data/standalone/entity/EntityTests.java

```diff
@@ -2442,8 +2442,10 @@ public void testUpdateQueryWithWhereClause() {
             // Ensure there is no data left over from other tests:
             shared.deleteIfPositive();
         } catch (UnsupportedOperationException x) {
-            if (type.isKeywordSupportAtOrBelow(DatabaseType.KEY_VALUE)) {
-                return; // Key-Value databases might not be capable of And.
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.GRAPH) && TestProperty.delay.isSet()) {
+                // NoSQL databases with eventual consistency might not be capable
+                // of counting removed entities.
+                // Use alternative approach for ensuring no data is present:
```

Comment:
Restore the skip for key-value databases that are not using eventual consistency:
```suggestion
            if (type.isKeywordSupportAtOrBelow(DatabaseType.KEY_VALUE)) {
                return; // Key-Value databases might not be capable of And.
            } else if (type.isKeywordSupportAtOrBelow(DatabaseType.GRAPH) && TestProperty.delay.isSet()) {
                // NoSQL databases with eventual consistency might not be capable
                // of counting removed entities.
                // Use alternative approach for ensuring no data is present:
```